### ISSUE TITLE
UI blurry

### ISSUE THREAD

**Issue by AJ-316** [praydog/REFramework#670]
The low resolution upscaling causes the UI to look blurry and has ghost trails.

**Comment by TeTr-a** [comment]
Perhaps worth noting that this also happens when applying DLAA, not just for DLSS upscaling

**Comment by Grammernatzi** [comment]
A tip I've found to help mitigate things is to turn on DLAA and not use DLSS settings, and instead change the image quality setting. At 1440p with 80% image quality, this still looks very good but the UI isn't nearly as blurry as quality mode DLSS. It's still a bit blurry, but it's probably the best solution for now.

**Comment by TeTr-a** [comment]
Thanks for the tip, this works well but still leaves the ghosting problem. I've found that the "Jitter" setting seems to remove the ghosting entirely, but then it seems like the AA isn't working at all. Tweaking the Jitter X and Y values produced some positive results for me.

**Comment by Grammernatzi** [comment]
What jitter X and Y worked the best for you? I haven't messed with it much yet as I find the current settings good; I have motion blur off, I don't know if you have it on.

**Comment by TeTr-a** [comment]
I've actually decided to just stick with the default jitter X and Y settings. changing them did remove ghosting, but it made the overall image very jagged. Motion Blur is disabled, and same with Bloom, which actually fixes the flickering lights issue present with the upscaler as well.

I hope it will be possible to fix the UI issues with the upscaler. REFramework already has options to disable/hide UI which makes me think that it shouldn't be too difficult to tell the upscaler to ignore the HUD. But I really have no idea, just hopeful! DLSS/DLAA is better than any of the options made available by Capcom, if this UI issue gets solved it is _perfect_.

**Comment by TeTr-a** [comment]
Good morning Gremmernetzi! I recommend you (and anyone else reading this thread) to download and use the latest version of "DLSSTweaks". 

It seems that RE Engine games don't play well with DLAA running at full resolution, so this tool will allow you to offset the resolution by -1, as show below.

![AEC3W9w](https://user-images.githubusercontent.com/112025198/227720902-78fa167d-c6b4-4ca7-8562-3d8ecf5e6a55.png)

This change has almost entirely removed any and all ghosting from RE4 for me. There is no longer a need to render at a lower image quality either, looks and feels fantastic. Hope this helps!

**Comment by Grammernatzi** [comment]
Well, I *want* to run at a lower image quality for better framerate, that's why I wanted to use DLSS, since I'm running at 1440p and want a higher framerate for my 144hz monitor. From what I'm seeing, this only applies to DLAA? But, the image quality thing should still work, I imagine.

**Comment by TeTr-a** [comment]
Yeah same, I want to use DLSS. Even at Quality level it's like a 40~50% increase in performance but the UI issues make it unusable for me. 

In case you didn't, you have to edit the dlsstweaks.ini and change the resolution offset manually.

**Comment by Grammernatzi** [comment]
I did, and I put the .dlls + ini in the folder. Dunno if there's anything else I need to change, but, oh well. I even did the signature override and restarted my PC after for good measure.

**Comment by TeTr-a** [comment]
Very sorry it did not work for you then :(

It didn't _completely_ remove the ghosting for me, but it is very very good now. 

**You have to be running dlss 3.1.1 or newer, 2.5.1 _will not work_ with DLSSTweaks.** 

I also edited the ini to force DLAA no matter what DLSS setting I have selected, pictured below.

![notepad++_5ZxzjoyMKk](https://user-images.githubusercontent.com/112025198/227765793-ca1376dc-dd09-4bf4-85e9-a19f3b533c28.png)

A good way to see if DLSSTweaks is working is to force DLAA through the ini, and then change the DLSS quality settings in REFramework; there should be no change in quality level.

**Comment by Grammernatzi** [comment]
Setting ForceDLAA helped. There's still some jitter but it's much better now. And now I don't have to worry about resetting it every time I boot up the game.

**Comment by krittinen** [comment]
> A tip I've found to help mitigate things is to turn on DLAA and not use DLSS settings, and instead change the image quality setting. At 1440p with 80% image quality, this still looks very good but the UI isn't nearly as blurry as quality mode DLSS. It's still a bit blurry, but it's probably the best solution for now. So far it looks pretty good: ![image](https://user-images.githubusercontent.com/869037/227688008-fc0d1945-02f0-4aca-90fc-99cfddfcec95.png)
> 
> If you're on AMD or Intel, I'm not sure if there's a similar setting for XeSS, but it's worth giving a shot I suppose.

Forcing DLAA does do the trick but then the resolution scaling setting doesn't work and it always renders at full res

**Comment by Grammernatzi** [comment]
That's why you turn down image quality in the game itself. DLAA will help upscale it.

**Comment by krittinen** [comment]
> That's why you turn down image quality in the game itself. DLAA will help upscale it.

Which is why I said, the ingame resolution scaler does not work when I have DLAA forced, it will still render at 100% scale.

**Comment by Grammernatzi** [comment]
It w

**Comment by Grammernatzi** [comment]
It seemed to work for me, I had a noticeable frame rate increase. I will check when I get back ingame later.

**Comment by Grammernatzi** [comment]
Huh, you are right, it seemed to stop working. I wonder if something changed that?

**Comment by krittinen** [comment]
> Huh, you are right, it seemed to stop working. I wonder if something changed that?

Wonder if it's a change in version, I'm using REFramework build **0eeadf2** and DLSS 3.1.11

**Comment by zaphellyn5467** [comment]
hope they fix this would be great🙂

**Comment by Filoppi** [comment]
Shouldn't the pd upscaler just hook to where Resident Evil 4 FSR is done and replace it with DLSS?
It seems like now it's done after post processing and UI, which is kind of a dead end of TAA, but the game native FSR implementation is certainly done before that. In many games it's possible to replace DLSS with FSR 2, so I guess we could do the opposite here.

**Comment by praydog** [comment]
> Shouldn't the pd upscaler just hook to where Resident Evil 4 FSR is done and replace it with DLSS? It seems like now it's done after post processing and UI, which is kind of a dead end of TAA, but the game native FSR implementation is certainly done before that. In many games it's possible to replace DLSS with FSR 2, so I guess we could do the opposite here.

Game's you're referring to with replaceable FSR are only because they come with the FSR DLL which is easily replaceable on disk. That's not the case here. The devs (or AMD...?) have decided to bake it into the executable itself, making replacing the implementation much harder and most likely moot, there is a lot of jitter/shimmer with FSR2 in this game (especially on foliage) which is not present when using this mod's version of FSR2.

**Comment by Filoppi** [comment]
All the issues would likely be fixed. Possibly it doesn't matter about the FSR dll being baked in, it would likely still be possible to hook into the engine upscaling calls and redirect them to DLSS instead.

**Comment by praydog** [comment]
UI just needs to be isolated. I already have a general idea of where this is happening, just more reverse engineering has to be done on how the game is rendering it to a specific render target.

**Comment by praydog** [comment]
A little update on this:

I've figured out how to isolate the UI (at least in SF6, it should transfer to RE4). This completely removes the jittering and ghosting on the UI, as well as the blurriness (though it seems like the fonts are still lower res). There are still some issues I'm working out with the color looking too dark on the composited UI, and issues with some depth-aware UI elements.

It's not released yet, but is looking promising.

**Comment by santiphap-luml** [comment]
Any updates?

**Comment by goat9988** [comment]
So any update about the UI blurry problems?

**Comment by B-Wilhelm** [comment]
My dog

**Comment by brad86m** [comment]
Patiently waiting on news for the UI fixing. DLAA is so lovely to have in this terrible engine, but the UI getting messed up does hurt the overall presentation and finish.

I really miss Capcom's MTFramework. Now that was one sublime engine.

**Comment by 13saint** [comment]
Ok. So I experimented with a lot of things to remedy the blurry UI and in the end, 4K DLSS Performance is the best mode for me with 1440p Native display (Yes, beating even 1440p DLAA). I use DLDSR 2.25x (which for 1440p is 4K), then I use DLSSTweaks to make sure Preset E is used (Preset E is considered by many to be the best preset for DLSS). When compared with other modes, 4K DLSS Performance beat 1440p DLSS Quality with no pb, but DLAA can be argued due to the fact that they are very similar in terms of image quality and UI imo. But the dealbreaker is that 4K DLSS Performance actually has more fps than 1440p DLAA, even Native res fps is no match.  In case it is not clear, I'm using 1440p display and thus is my native resolution.

**Comment by AJ-316** [comment]
### **For anyone it still is not working**

Try running the game in borderless window, that somehow nearly fixes it

**Comment by VladimirGolosin** [comment]
@praydog have you managed to do it?

**Comment by qvattr0** [comment]
@praydog Any updates on isolating the UI?

**Comment by shibeh** [comment]
Would love to get an update on this too!

**Comment by he2a** [comment]
@praydog Its now been 2 years. Eagerly waiting for the update!

**Comment by G4judge** [comment]
@praydog any updates ? cause GTA 6 will release before you fix it

**Comment by sikicila01** [comment]
no fix yet?

**Comment by PugsleytheFool** [comment]
Any update on the UI fix? Honestly RE4 otherwise looks perfect with DLSS once you simply disable bloom, so it would be incredible to have the UI not ghost and jitter!

**Comment by VanitasFR** [comment]
Any updates ?

**Comment by G4judge** [comment]
Nope for some reason he only updated reframework for re9

**Comment by Taiyaki777** [comment]
Wish these issues (ghosting, and annoying unstable subtitles and UI) would get fixed, cause this would turn into the go to option for everyone. Instead it's a bit of pick your poison. Still thank you for all the incredible hard work put into REframework.

**Comment by SercanDmr** [comment]
Especially steam deck and other handhelds suffer from this most.

I hope someday this issue gets fixed...
